diff -Nur jpeg-6b.orig/makefile.cfg jpeg-6b/makefile.cfg
--- jpeg-6b.orig/makefile.cfg	1998-03-22 03:08:57.000000000 +0800
+++ jpeg-6b/makefile.cfg	2010-01-01 20:32:59.000000000 +0800
@@ -17,7 +17,6 @@
 binprefix =
 manprefix =
 manext = 1
-mandir = $(prefix)/man/man$(manext)
 
 # The name of your C compiler:
 CC= @CC@
@@ -196,11 +195,6 @@
 	$(INSTALL_PROGRAM) jpegtran $(bindir)/$(binprefix)jpegtran
 	$(INSTALL_PROGRAM) rdjpgcom $(bindir)/$(binprefix)rdjpgcom
 	$(INSTALL_PROGRAM) wrjpgcom $(bindir)/$(binprefix)wrjpgcom
-	$(INSTALL_DATA) $(srcdir)/cjpeg.1 $(mandir)/$(manprefix)cjpeg.$(manext)
-	$(INSTALL_DATA) $(srcdir)/djpeg.1 $(mandir)/$(manprefix)djpeg.$(manext)
-	$(INSTALL_DATA) $(srcdir)/jpegtran.1 $(mandir)/$(manprefix)jpegtran.$(manext)
-	$(INSTALL_DATA) $(srcdir)/rdjpgcom.1 $(mandir)/$(manprefix)rdjpgcom.$(manext)
-	$(INSTALL_DATA) $(srcdir)/wrjpgcom.1 $(mandir)/$(manprefix)wrjpgcom.$(manext)
 
 install-lib: libjpeg.$(A) install-headers
 	$(INSTALL_LIB) libjpeg.$(A) $(libdir)/$(binprefix)libjpeg.$(A)
