diff -Nur fbv-1.0b.orig/Makefile fbv-1.0b/Makefile
--- fbv-1.0b.orig/Makefile	2004-09-07 20:34:29.000000000 +0800
+++ fbv-1.0b/Makefile	2010-01-01 20:13:44.000000000 +0800
@@ -29,8 +29,6 @@
 
 install: $(OUT)
 	cp $(OUT) $(DESTDIR)$(bindir)
-	gzip -9c $(OUT).1 > $(DESTDIR)$(mandir)/man1/$(OUT).1.gz
 
 uninstall: $(DESTDIR)$(bindir)/$(OUT)
 	rm -f $(DESTDIR)$(bindir)/$(OUT)
-	rm -f $(DESTDIR)$(mandir)/man1/$(OUT).1.gz
